cmd/virtual-workspaces/command/cmd.go

```diff
@@ -178,6 +183,18 @@ func Run(ctx context.Context, o *options.Options) error {
 		return err
 	}
 
+	// apply admission to check permission claims for resources through a view's URL
+	admissionPluginInitializers := []admission.PluginInitializer{
+		kcpadmissioninitializers.NewKcpInformersInitializer(wildcardKcpInformers, cacheKcpInformers),
+	}
+	o.VirtualWorkspaces.Admission.ApplyTo(
+		&recommendedConfig.Config,
+		informerfactoryhack.Wrap(wildcardKubeInformers),
+		clientsethack.Wrap(localShardKubeClusterClient),
```

Comment:
i am getting nervous with all those hacks 😄

Reply:
Copied forward from https://github.com/kcp-dev/kubernetes/blob/6ebb9f064d8cd524bafba7154d5e47619303be0a/pkg/genericcontrolplane/server.go#L159-L166. Not net new. It's to be able to pass cluster-scoped informer factories/clients to admission that only expects normal single-cluster factories/clients.